### ISSUE TITLE
Allow shorthand syntax for SSH URIs

### ISSUE THREAD

**Issue by olafmustafar** [cpm-cmake/CPM.cmake#491]
I'm trying to add an internal repository using the ssh protocol:
I know it will work if write
```cmake
 CPMAddPackage(
     GIT_REPOSITORY "git@github.com:xxxxxx/yyyyyy.git"
     GIT_TAG master
 )
```
I think it would be nice if it could be done using oneline like this:
```cmake
  CPMAddPackage( "git@github.com:xxxxxx/yyyyyy.git#master")
  #or
  CPMAddPackage( "ssh:git@github.com:xxxxxx/yyyyyy.git#master")
```
Is it possible? 
// I took a look into the code and know there can be some issues parsing the '@' char, maybe some improvments on the regex could fix this.

**Comment by TheLartians** [comment]
One possibility to use the shorthand syntax in the current version of CPM would be to locally configure git to use ssh instead of https for all Github requests. E.g. by running

```bash
git config --global url."ssh://git@github.com".insteadOf "https://github.com"
```

**Comment by olafmustafar** [comment]
Yes, this would work. But its preferable if i could use the shorthand syntax without changing the global configurations
